pkg/sql/plan2/build_subquery.go

```diff
@@ -63,25 +47,40 @@ func buildSubQuery(subquery *tree.Subquery, ctx CompilerContext, query *Query, s
 			return nil, err
 		}
 	case *tree.SelectClause:
-		// buildSelect(selectClause.Select, ctx, query)
 		return nil, errors.New(errno.SQLStatementNotYetComplete, fmt.Sprintf("support select statement: %T", subquery))
 	default:
 		return nil, errors.New(errno.SQLStatementNotYetComplete, fmt.Sprintf("unknown select statement: %T", subquery))
 	}
-
 	expr.IsCorrelated = newCtx.subQueryIsCorrelated
 
-	//把子查询的节点放到前面去
-	query.Steps = []int32{int32(nowLength - 1)}
-	query.Nodes = append(query.Nodes[nowLength:], query.Nodes[:nowLength]...)
+	//move subquery node to top
+	for i := len(query.Nodes) - 1; i >= 0; i-- {
+		if query.Nodes[i].NodeId == nodeId {
+			query.Nodes = append(query.Nodes[i+1:], query.Nodes[:i+1]...)
+			break
+		}
```

Comment:
Moving nodes around is error-prone. And no need to do that.

Reply:
sometime when you will append something to PreNode,  If the pre node is subquery, that will be error:
eg: select a from tbl where b>(subquery) limit 10
I will improve the code after test case cover more corner